code_map.py

```diff
@@ -787,41 +774,35 @@ def run(self, edit):
                     code_map_group = create_codemap_group()
                     CodeMapListener.map_group = code_map_group
 
-            with open(code_map_file(), "w") as file:
+            with open(code_map_file, "w") as file:
                 file.write('')
 
-            map_view = w.open_file(code_map_file())
-            map_view.settings().set("margin", 8)
+            map_view = w.open_file(code_map_file)
+
+            # default margin: 8
+            map_view.settings().set("margin", settings().get('codemap_margin', 8))
+
+            # allow custom font face/size, it's optional and it doesn't need to be in settings
+            if settings().has('codemap_font_size'):
+                map_view.settings().set("font_size", settings().get('codemap_font_size'))
+            if settings().has('codemap_font_face'):
+                map_view.settings().set("font_face", settings().get('codemap_font_face'))
+
             map_view.settings().set("word_wrap", False)
             map_view.settings().set("gutter", False)
             map_view.settings().set("draw_white_space", "none")
             w.set_view_index(map_view, code_map_group, 0)
             map_view.sel().clear()
 
-            is_in_same_group = (code_map_group == get_group(current_view))
-
-            if not is_in_same_group:
-                # cannot use CodeMapListener.active_view as it will be immediately
-                # overwritten by other views from the new group being activated
-                w.focus_view(current_view)
-            else:
-                w.focus_view(map_view)
-                if code_map_group > 0: # move focus to the left group to trigger map refresh
-                    w.focus_group(code_map_group-1)
+            w.run_command("synch_code_map")
+            focus_source_code()
 
-        else:                       # closing Code Map
-            is_map_view_active = w.active_view_in_group(get_group(map_view)) == map_view
-            is_in_same_group = get_group(map_view) == get_group(current_view)
```

Comment:
For this I am in doubt, you probably did it for a reason, but I can't see it. It seems much simpler to me like this, but I don't know exactly why you were doing this.

Reply:
It was a very specific case of some conflict with "Favorites" view. I think it was something about closing/not closing the group.

Though it seems like your change to the "on-open" routine has achieved the same outcome.